server/src/main/java/org/opensearch/indices/TieredCacheSpilloverStrategyHandler.java

```diff
@@ -132,15 +131,24 @@ private Function<K, CacheValue<V>> getValueFromTierCache() {
         return key -> {
             for (CachingTier<K, V> cachingTier : cachingTierList) {
                 V value = cachingTier.get(key);
+                double getTimeEWMA = getTimeEWMAIfDisk(cachingTier);
                 if (value != null) {
-                    tieredCacheEventListener.onHit(key, value, cachingTier.getTierType());
+                    tieredCacheEventListener.onHit(key, value, cachingTier.getTierType(), getTimeEWMA);
                     return new CacheValue<>(value, cachingTier.getTierType());
                 }
-                tieredCacheEventListener.onMiss(key, cachingTier.getTierType());
+                tieredCacheEventListener.onMiss(key, cachingTier.getTierType(), getTimeEWMA);
```

Comment:
This doesn't seem right. We should ideally put these get times inside Disk caching tier itself. So that if we have a different implementation of TieredService, we don't have to duplicate this work.

Reply:
And as discussed having separate DiskCacheStats separately should be able to solve this.